File: 19/Go/deviceid.go
package main

import (
	"crypto/hmac"
	"crypto/rand"
	"crypto/sha1"
	"encoding/hex"
	"fmt"
)

func generateDeviceId() string {
	identifier := make([]byte, 20)
	_, err := rand.Read(identifier)
	if err != nil {
		fmt.Println("Error: ", err)
	}

	key, err := hex.DecodeString("E7309ECC0953C6FA60005B2765F99DBBC965C8E9")
	if err != nil {
		fmt.Println("Error: ", err)
	}

	h := hmac.New(sha1.New, key)
	h.Write(append([]byte{0x19}, identifier...))
	return fmt.Sprintf("19%x%x", identifier, h.Sum(nil))
}

func main() {
	fmt.Println(generateDeviceId())
}

File: 19/Python/deviceId.py
import os
import hmac
from hashlib import sha1


def generate_device_id() -> str:
    identifier = os.urandom(20)
    key = bytes.fromhex("E7309ECC0953C6FA60005B2765F99DBBC965C8E9")
    mac = hmac.new(key, b"\x19" + identifier, sha1)
    return f"19{identifier.hex()}{mac.hexdigest()}"


print(generate_device_id())